src/transformers/pipelines/visual_question_answering.py

```diff
@@ -0,0 +1,165 @@
+        - `pipeline(image=[image, image], question=[question])`
+        - `pipeline(image=[image], question=[question, question])`
+        - `pipeline(image=[image], question=[question, question])`
+        - `pipeline({"image": image, "question": question})`
+        - `pipeline([{"image": image, "question": question}])`
+        - `pipeline([{"image": image, "question": question}, {"image": image, "question": question}])`
+
+        Args:
+            image (`str`, `List[str]`, `PIL.Image` or `List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing a http link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images. If given a single image, it can be
+                broadcasted to multiple questions.
+            question (`str`, `List[str]`):
+                The question(s) asked. If given a single question, it can be broadcasted to multiple images.
+            top_k (`int`, *optional*, defaults to 5):
+                The number of top labels that will be returned by the pipeline. If the provided number is higher than
+                the number of labels available in the model configuration, it will default to the number of labels.
+        Return:
+            A dictionary or a list of dictionaries containing result. The dictionaries contain the following keys:
+
+            - **label** (`str`) -- The label identified by the model.
+            - **score** (`int`) -- The score attributed by the model for that label.
+        """
+
+        pipeline_inputs = self._args_parser(*args, **kwargs)
+        results = super().__call__(pipeline_inputs, **kwargs)
+        return results
+
+    def preprocess(self, pipeline_input, padding=False, truncation=False):
+        image = load_image(pipeline_input["image"])
+        model_inputs = self.tokenizer(
+            pipeline_input["question"], return_tensors=self.framework, padding=padding, truncation=truncation
+        )
+        image_features = self.feature_extractor(images=image, return_tensors=self.framework)
+        model_inputs.update(image_features)
+        return model_inputs
+
+    def _forward(self, model_inputs):
+        model_outputs = self.model(**model_inputs)
+        return model_outputs
+
+    def postprocess(self, model_outputs, top_k=5):
+        if top_k > self.model.config.num_labels:
+            top_k = self.model.config.num_labels
+
+        if self.framework == "pt":
+            probs = model_outputs.logits.softmax(-1)[0]
+            scores, ids = probs.topk(top_k)
+        else:
+            raise ValueError(f"Unsupported framework: {self.framework}")
+
+        scores = scores.tolist()
+        ids = ids.tolist()
+        return [{"score": score, "label": self.model.config.id2label[_id]} for score, _id in zip(scores, ids)]
```

Comment:
Hmm in the future, there may also be generative models for VQA (that simply generate the answer in text). 

ViLT is a discriminative model (a multi-label classifier), but we need to keep this in mind for the future.

Reply:
Yeah I agree. Seems like it's a common practice to format the VQA as a classification task in the academic setup but text generation is definite more applicable in the real world. I don't have a good answer for this now. Maybe we can figure it out when there is a VQA model with a Casual LM head.